zarr/convenience.py

```diff
@@ -949,7 +943,7 @@ def _copy(log, source, dest, name, root, shallow, without_attrs, if_exists, dry_
         if do_copy:
 
             # log a message about what we're going to do
-            log("copy {} {} {}".format(source.name, source.shape, source.dtype))
```

Comment:
Perhaps `log("{}".format(arg))` should actually be changed to `log("%s", arg)`, so that string formatting interpolation is not executed if logging is not enabled.
* Some developers argue that deferring that string interpolation in the logging function, after checking whether logging is enabled, is a perceptible speed win if logging is disabled. See Pylint's [logging-fstring-interpolation / W1203](https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html).
* Other developers argue that the speed win is hardly perceptible because the cost of a function call is much higher, while the speed win of f-strings if logging is enabled is more perceptible. See https://github.com/python/cpython/issues/90358.

See also:
* https://docs.python.org/3/howto/logging.html#optimization